apps/admin/frontend/src/api.ts

```diff
@@ -102,3 +104,265 @@ export const unprogramCard = {
     });
   },
 } as const;
+
+// Queries
+
+type QueryProps<Method extends keyof ApiClient> = Parameters<
+  ApiClient[Method]
+>[0];
+
+export const getCurrentElectionMetadata = {
+  queryKey(): QueryKey {
+    return ['getCurrentElectionMetadata'];
+  },
+  useQuery() {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(), () =>
+      apiClient.getCurrentElectionMetadata()
+    );
+  },
+} as const;
+
+export const getCastVoteRecordFiles = {
+  queryKey(): QueryKey {
+    return ['getCastVoteRecordFiles'];
+  },
+  useQuery() {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(), () => apiClient.getCastVoteRecordFiles());
+  },
+} as const;
+
+export const getCastVoteRecords = {
+  queryKey(): QueryKey {
+    return ['getCastVoteRecords'];
+  },
+  useQuery() {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(), () => apiClient.getCastVoteRecords());
+  },
+} as const;
+
+export const getCastVoteRecordFileMode = {
+  queryKey(): QueryKey {
+    return ['getCastVoteRecordFileMode'];
+  },
+  useQuery() {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(), () =>
+      apiClient.getCastVoteRecordFileMode()
+    );
+  },
+} as const;
+
+type GetWriteInsProps = QueryProps<'getWriteIns'>;
+export const getWriteIns = {
+  queryKey(props?: GetWriteInsProps): QueryKey {
+    return props ? ['getWriteIns', props] : ['getWriteIns'];
```

Comment:
The `['getWriteIns']` variant is important for when you're using the `queryKey` to invalidate queries. I don't think `['getWriteIns', undefined]` would clear the query. But, we could special case the query invalidation differently by either making a specific method for it, or a specific property with the query root `'getWriteIns'`.

Reply:
I think it would work with [`getWriteIns`, undefined]` since the query key matching uses prefixes: https://tanstack.com/query/v4/docs/react/guides/query-invalidation#query-matching-with-invalidatequeries. But I could be wrong

Reply:
OK, I tested and it did not work. My understanding of that reference page is that `['getWriteIns']` would operate as a prefix for `['getWriteIns', anything]` but that `['getWriteIns, undefined]` is not a prefix for `['getWriteIns', 'key']`.

Reply:
Yeah that makes sense. Thanks for testing it out!